UI Crash in Download UI on screen rotation

### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.1

### Steps to reproduce the bug

1. Click on any video
2. Click download
3. Turn/Rotate phone 90 degrees
4. The UI Crashes

### Expected behavior

Doesn't crash

### Actual behavior

It crashes

### Screenshots/Screen recordings

<img alt="Download UI" src="https://user-images.githubusercontent.com/102857306/179365500-ba285662-cc5c-43ef-92e8-55d19936a56b.png" height=600 />
Screenshot to help you identify what I meant by Download UI.
That screen crashes when rotation phone screen.

### Logs

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ DE
* __Content Language:__ de-DE
* __App Language:__ de_DE
* __Service:__ none
* __Version:__ 0.23.1
* __OS:__ Linux Android 7.0 - 24
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.schabi.newpipe/org.schabi.newpipe.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2793)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2864)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4720)
	at android.app.ActivityThread.-wrap20(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1573)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:156)
	at android.app.ActivityThread.main(ActivityThread.java:6517)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)
Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:628)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:483)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
	at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2728)
	at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:198)
	at androidx.fragment.app.FragmentActivity$2.onContextAvailable(FragmentActivity.java:149)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:322)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:273)
	at org.schabi.newpipe.MainActivity.onCreate(MainActivity.java:143)
	at android.app.Activity.performCreate(Activity.java:6915)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1123)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2746)
	... 10 more
Caused by: java.lang.NoSuchMethodException: <init> []
	at java.lang.Class.getConstructor0(Class.java:2204)
	at java.lang.Class.getConstructor(Class.java:1683)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:613)
	... 23 more

```
</details>
<hr>


### Affected Android/Custom ROM version

Android 7.0

### Affected device model

Huawei P9 (EVA-L09)

### Additional information

_No response_

I can reproduce this bug in 0.23.2 on Android 9 - 28, was just about to file it:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.schabi.newpipe/org.schabi.newpipe.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2974)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3109)
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:4904)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4813)
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1834)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6826)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:628)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:483)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
	at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2728)
	at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:198)
	at androidx.fragment.app.FragmentActivity$2.onContextAvailable(FragmentActivity.java:149)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:322)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:273)
	at org.schabi.newpipe.MainActivity.onCreate(MainActivity.java:143)
	at android.app.Activity.performCreate(Activity.java:8136)
	at android.app.Activity.performCreate(Activity.java:8127)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1272)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2954)
	... 13 more
Caused by: java.lang.NoSuchMethodException: <init> []
	at java.lang.Class.getConstructor0(Class.java:2327)
	at java.lang.Class.getConstructor(Class.java:1725)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:613)
	... 27 more

```

@opusforlife2 bug is still reproducible in version 0.23.2 like @GfEW said. I opened the Downloads UI by clicking on the icon on the right side below the video title in the app. (I think that means sidebar?)
formatted crash log below:

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ DE
* __Content Language:__ de-DE
* __App Language:__ de_DE
* __Service:__ none
* __Version:__ 0.23.2
* __OS:__ Linux Android 7.0 - 24
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.schabi.newpipe/org.schabi.newpipe.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2793)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2864)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4720)
	at android.app.ActivityThread.-wrap20(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1573)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:156)
	at android.app.ActivityThread.main(ActivityThread.java:6517)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)
Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment org.schabi.newpipe.download.DownloadDialog: could not find Fragment constructor
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:628)
	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:483)
	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
	at androidx.fragment.app.FragmentManager.restoreSaveState(FragmentManager.java:2728)
	at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:198)
	at androidx.fragment.app.FragmentActivity$2.onContextAvailable(FragmentActivity.java:149)
	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:322)
	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:273)
	at org.schabi.newpipe.MainActivity.onCreate(MainActivity.java:143)
	at android.app.Activity.performCreate(Activity.java:6915)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1123)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2746)
	... 10 more
Caused by: java.lang.NoSuchMethodException: <init> []
	at java.lang.Class.getConstructor0(Class.java:2204)
	at java.lang.Class.getConstructor(Class.java:1683)
	at androidx.fragment.app.Fragment.instantiate(Fragment.java:613)
	... 23 more

```
</details>
<hr>

--- a/app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java
+++ b/app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java
@@ -145,6 +145,12 @@ public class DownloadDialog extends DialogFragment
     // Instance creation
     //////////////////////////////////////////////////////////////////////////*/
 
+    public DownloadDialog() {
+        // Just an empty default no-arg ctor to keep Fragment.instantiate() happy
+        // otherwise InstantiationException will be thrown when fragment is recreated
+        // TODO: Maybe use a custom FragmentFactory instead?
+    }
+
     /**
      * Create a new download dialog with the video, audio and subtitle streams from the provided
      * stream info. Video streams and video-only streams will be put into a single list menu,

